clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -121,21 +133,41 @@ public FSDataOutputStream create(Path cPath, FsPermission permission, boolean ov
     }
 
     TachyonURI path = new TachyonURI(Utils.getPathWithoutScheme(cPath));
-    if (mTFS.exist(path)) {
-      if (overwrite && !mTFS.getFileStatus(-1, path).isFolder) {
-        if (!mTFS.delete(path, false)) {
-          throw new IOException("Failed to delete existing data " + cPath);
+    try {
+      TachyonFile file = mTFS.open(path);
+      if (overwrite) {
+        FileInfo info = mTFS.getInfo(file);
+        if (info.isIsFolder()) {
+          throw new IOException(cPath.toString() + " already exists. Directories cannot be "
+              + "overwritten with create.");
+        } else {
+          try {
+            mTFS.delete(file);
+          } catch (TachyonException e) {
+            throw new IOException("Failed to delete existing data " + cPath, e);
+          }
         }
       } else {
-        throw new IOException(cPath.toString() + " already exists. Directories cannot be "
-            + "overwritten with create.");
+        throw new IOException(cPath.toString() + " already exists.");
       }
+    } catch (InvalidPathException e) {
+      // The file doesn't exist, fall through to create it.
+    } catch (FileDoesNotExistException e) {
+      // The file doesn't exist, fall through to create it.
+    } catch (TachyonException e) {
+      throw new IOException(e);
+    }
+
+    OutStreamOptions options =
+        new OutStreamOptions.Builder(mTachyonConf).setBlockSizeBytes(blockSize).build();
+    FileOutStream outStream;
+    try {
+      outStream = mTFS.getOutStream(path, options);
+    } catch (TachyonException e) {
+      throw new IOException(e);
     }
-    long fileId = mTFS.createFile(path, blockSize);
-    TachyonFile file = mTFS.getFile(fileId);
-    file.setUFSConf(getConf());
 
-    return new FSDataOutputStream(file.getOutStream(), mStatistics);
+    return new FSDataOutputStream(outStream, mStatistics);
```

Comment:
Perhaps you can just do this in the `try catch`? I notice you use a different style in the `delete` method (return from the try catch)